### DISCUSSION TITLE
Prerendered bulid with NX problems

### DISCUSSION THREAD

**Discussion by aldrio** [vikejs/vike#1246] [Discussion]
Hello, i'm trying to get SSG working in my NX based project. When I build my webapp from the command line with vite directly vike works fine, however I'm having trouble getting the build with prerendering to work using the nx executor.

I've found I have to add `disableAutoFullBuild: false` to the config it gets to the prerendering step, but then it still crashes with `Error: Cannot find module '../assets.json'`

I traced the issue to here:
https://github.com/vikejs/vike/blob/4ce0c180b63e44397de8a7ff2b53c65d5c69d4e8/vike/node/plugin/plugins/autoFullBuild.ts#L70-L78
The options passed to vite build from the nx executor, including options about the correct output build path, aren't correctly passed through because the code is only passing options if it was a command line call. Locally i made a patch to change that selection to be `const configInline = config.inlineConfig;` So I guess I'm wondering if that makes sense or would have any problems? I'm not very familiar with vite I just happened to see inlineConfig when checking stuff out and it seems to work. I'm assuming either config.inlineConfig didn't exist at the time this was written or there's a reason for not using it

**Comment by brillout** [comment]
> Locally i made a patch to change that selection to be `const configInline = config.inlineConfig;`

👍 What exactly are these changes you made? (Feel free to publish your changes as a PR.)

**Comment by shishkin17** [comment]
I also had problems with launching prerendering in nx environment and this issue helped me to solve it where I accidentally found the existence of disableAutoFullBuild option.

@brillout you should definitely add the description of disableAutoFullBuild to docs.
Thank you for your work!

**Reply by brillout** [reply]
Why do you need `disableAutoFullBuild`?

> add [...] disableAutoFullBuild to docs.

We're working on the Vike CLI which will have the options `$ vike build --client`, `$ vike build --server`, and `$ vike build --skip-prerender`. That will do the trick, I guess.

**Reply by shishkin17** [reply]
Without `disableAutoFullBuild: false` prerendering doesn't start when using nx executors to build a project.
